hw/chisel/src/main/scala/snax/streamer/StreamerTop.scala

```diff
@@ -0,0 +1,237 @@
+
+}
+
+// Scala main function for generating test streamerTop system verilog file
+object StreamerTopTester extends App {
+  emitVerilog(
+    new StreamerTop(
+      StreamerParams(
+        temporalAddrGenUnitParams =
+          StreamerTestConstant.temporalAddrGenUnitParams,
+        fifoReaderParams = StreamerTestConstant.fifoReaderParams,
+        fifoWriterParams = StreamerTestConstant.fifoWriterParams,
+        stationarity = StreamerTestConstant.stationarity,
+        dataReaderParams = StreamerTestConstant.dataReaderParams,
+        dataWriterParams = StreamerTestConstant.dataWriterParams
+      )
+    ),
+    Array("--target-dir", "generated/streamertop/tester")
+  )
+}
+
+// streamertop for GEMM
+object GeMMStreamerTop {
+  def main(args: Array[String]): Unit = {
+    val outPath = args.headOption.getOrElse("generated/streamertop/gemm")
+    emitVerilog(
+      new StreamerTop(
+        StreamerParams(
+          temporalAddrGenUnitParams =
+            GeMMStreamerParameters.temporalAddrGenUnitParams,
+          fifoReaderParams = GeMMStreamerParameters.fifoReaderParams,
+          fifoWriterParams = GeMMStreamerParameters.fifoWriterParams,
+          stationarity = GeMMStreamerParameters.stationarity,
+          dataReaderParams = GeMMStreamerParameters.dataReaderParams,
+          dataWriterParams = GeMMStreamerParameters.dataWriterParams,
+          tagName = "GeMM"
+        )
+      ),
+      Array("--target-dir", outPath)
+    )
+  }
+}
+
+// streamertop for PP-SIMD
+object PostProcessingStreamerTop {
+  def main(args: Array[String]): Unit = {
+    val outPath = args.headOption.getOrElse("generated/streamertop/simd")
+    emitVerilog(
+      new StreamerTop(
+        StreamerParams(
+          temporalAddrGenUnitParams =
+            PostProcessingStreamerParameters.temporalAddrGenUnitParams,
+          fifoReaderParams = PostProcessingStreamerParameters.fifoReaderParams,
+          fifoWriterParams = PostProcessingStreamerParameters.fifoWriterParams,
+          stationarity = PostProcessingStreamerParameters.stationarity,
+          dataReaderParams = PostProcessingStreamerParameters.dataReaderParams,
+          dataWriterParams = PostProcessingStreamerParameters.dataWriterParams,
+          tagName = "PostProcessingSIMD"
+        )
+      ),
+      Array("--target-dir", outPath)
+    )
+  }
+}
+
+// streamertop for MAC
+object MacStreamerTop extends App {
+  emitVerilog(
+    new StreamerTop(
+      StreamerParams(
+        temporalAddrGenUnitParams =
+          MacStreamerParameters.temporalAddrGenUnitParams,
+        fifoReaderParams = MacStreamerParameters.fifoReaderParams,
+        fifoWriterParams = MacStreamerParameters.fifoWriterParams,
+        stationarity = MacStreamerParameters.stationarity,
+        dataReaderParams = MacStreamerParameters.dataReaderParams,
+        dataWriterParams = MacStreamerParameters.dataWriterParams
+      )
+    ),
+    Array("--target-dir", "generated/streamertop/mac")
+  )
+}
```

Comment:
duplicate code as in streamer.scala: is one of both useless? Where would `object GemmStreamer` from `Streamer.scala` be used?
can we implement them in the source code of the respective modules?

Reply:
> duplicate code as in streamer.scala: is one of both useless? Where would `object GemmStreamer` from `Streamer.scala` be used? 

One is for instantiating the streamer without CsrManager and one is with CsrManager. For me, leaving them is ok.

> can we implement them in the source code of the respective modules?

Answered before.

Reply:
> One is for instantiating the streamer without CsrManager and one is with CsrManager. For me, leaving them is ok.

Why would we need one without a CsrManager?

Reply:
For testing purposes. We don't have CsrManager at the beginning so I generated and tested standalone Streamer. It would be useful for situations when I only modify the Streamer and want to generate and test it standalone later.

Reply:
If it is for testing purposes, please put it in a test. 
If it is for integration in another module, please put it in the other module.
Otherwise the code is scattered across files and unmaintainable. We do not want the streamer configuration for all existing accelerators to come in this file and `Streamer.scala`, right?

Reply:
It is a main function for testing if the streamer generates correctly under different configurations. So it is fine to put it here and it is not scattered but quite centered...

Reply:
But why is there testing code in the source code? Why can this not be a test file? This is very confusing

Reply:
I understand there is a need for emitVerilog statements, but here, half of them are unused, the streamer modules now contain parameters of different systems which have nothing to do with the streamers, and I truly think this is just bad design. The parameters for gemm streamers should be under the gemm folder, not the streamer folder. 

This is how it should be done:

```
hw/chisel/.../streamertop:
    package snax.streamer:
    export StreamerTop

hw/gemm_accelerator/chisel/...
    import snax.streamer.StreamerTop
    GeMMStreamer = emitVerilog(StreamerTop(GemmParams))
    
hw/other_accelerator/chisel/...
    import snax.streamer.StreamerTop
    OtherAcceleratorParams = emitVerilog(StreamerTop(OtherAcceleratorParams))
```

If it is not done like this, we're breaking a lot of rules regarding the directory structure, and we're doing the opposite of cleaning up the repository.

What is the advantage of the current approach? I really don't see this?